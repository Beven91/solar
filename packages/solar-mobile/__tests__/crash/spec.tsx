import React from 'react';
import Crash from '../../src/crash';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';


describe('Crash', () => {
  it('no message', () => {
    const tree = mount(
      <Crash title="no-message" message="您还没有登录" />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders not-found', () => {
    const tree = mount(
      <Crash title="not-found" />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });


  it('renders unlogin', () => {
    const tree = mount(
      <Crash title="unlogin" message="您还没有登录" />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });


  it('renders default', () => {
    const tree = mount(
      <Crash message={null} />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('renders custom', () => {
    mount(
      <Crash message={'自定义图片'} icon="ASSSDF" title="unlogin" />
    );
  });

  it('renders no button', () => {
    const tree = mount(
      <Crash onClick={null} message={'自定义图片'} icon="ASSSDF" />
    );

    expect(toJson(tree)).toMatchSnapshot();
  });
});
