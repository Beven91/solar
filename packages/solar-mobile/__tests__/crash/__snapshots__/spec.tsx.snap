// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crash no message 1`] = `
<Crash
  btnText="返回"
  className=""
  icon=""
  message="您还没有登录"
  onClick={[Function]}
  title="no-message"
>
  <div
    className="crash-component  "
  >
    <img
      className="logo"
      src=""
    />
    <div
      className="crash-title"
    >
       
      no-message
    </div>
    <div
      className="crash-description"
    >
      您还没有登录
    </div>
    <div
      className="crash-button"
      onClick={[Function]}
    >
       
      返回
       
    </div>
  </div>
</Crash>
`;

exports[`Crash renders default 1`] = `
<Crash
  btnText="返回"
  className=""
  icon=""
  message={null}
  onClick={[Function]}
  title="页面竟然崩溃了"
>
  <div
    className="crash-component  "
  >
    <img
      className="logo"
      src=""
    />
    <div
      className="crash-title"
    >
       
      页面竟然崩溃了
    </div>
    <div
      className="crash-button"
      onClick={[Function]}
    >
       
      返回
       
    </div>
  </div>
</Crash>
`;

exports[`Crash renders no button 1`] = `
<Crash
  btnText="返回"
  className=""
  icon="ASSSDF"
  message="自定义图片"
  onClick={null}
  title="页面竟然崩溃了"
>
  <div
    className="crash-component  "
  >
    <img
      className="logo"
      src="ASSSDF"
    />
    <div
      className="crash-title"
    >
       
      页面竟然崩溃了
    </div>
    <div
      className="crash-description"
    >
      自定义图片
    </div>
  </div>
</Crash>
`;

exports[`Crash renders not-found 1`] = `
<Crash
  btnText="返回"
  className=""
  icon=""
  message="大侠别急，您可点击以返回试试~"
  onClick={[Function]}
  title="not-found"
>
  <div
    className="crash-component  "
  >
    <img
      className="logo"
      src=""
    />
    <div
      className="crash-title"
    >
       
      not-found
    </div>
    <div
      className="crash-description"
    >
      大侠别急，您可点击以返回试试~
    </div>
    <div
      className="crash-button"
      onClick={[Function]}
    >
       
      返回
       
    </div>
  </div>
</Crash>
`;

exports[`Crash renders unlogin 1`] = `
<Crash
  btnText="返回"
  className=""
  icon=""
  message="您还没有登录"
  onClick={[Function]}
  title="unlogin"
>
  <div
    className="crash-component  "
  >
    <img
      className="logo"
      src=""
    />
    <div
      className="crash-title"
    >
       
      unlogin
    </div>
    <div
      className="crash-description"
    >
      您还没有登录
    </div>
    <div
      className="crash-button"
      onClick={[Function]}
    >
       
      返回
       
    </div>
  </div>
</Crash>
`;
